src/main/java/org/prebid/server/settings/service/JdbcPeriodicRefreshService.java

```diff
@@ -0,0 +1,199 @@
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * <p>
+ * Service that periodically calls database for stored request updates.
+ * If refreshRate is negative, then the data will never be refreshed.
+ * <p>
+ * The Queries should return a ResultSet with the following columns and types:
+ * <pre>
+ * 1. id: string
+ * 2. data: JSON
+ * 3. type: string ("request" or "imp")
+ * </pre>
+ *
+ * <p>
+ * If data is empty or the JSON "null", then the ID will be invalidated (e.g. a deletion).
+ * If data is not empty, depending on TYPE, it should be put to corresponding map with ID as a key and DATA as value.
+ * </p>
+ */
+public class JdbcPeriodicRefreshService {
+
+    private static final Logger logger = LoggerFactory.getLogger(JdbcPeriodicRefreshService.class);
+
+    private final CacheNotificationListener cacheNotificationListener;
+    private final Vertx vertx;
+    private final JdbcClient jdbcClient;
+    private final long refreshPeriod;
+
+    /**
+     * Example of initialize query:
+     * <pre>
+     * SELECT id, requestData, type
+     * FROM stored_requests;
+     * <pre>
+     * This query will be run once on startup to fetch _all_ known Stored Request data from the database.
+     */
+    private final String initQuery;
+
+    /**
+     * Example of initialize query:
+     * <pre>
+     * SELECT id, requestData, type
+     * FROM stored_requests
+     * WHERE last_updated > ?;
+     * <pre>
+     * The code will be run periodically to fetch updates from the database.
+     * Wildcard "?" would be used to pass last update date automatically.
+     */
+    private final String updateQuery;
+    private final TimeoutFactory timeoutFactory;
+    private final long timeout;
+    private Instant lastUpdate;
+
+    public JdbcPeriodicRefreshService(CacheNotificationListener cacheNotificationListener,
+                                      Vertx vertx, JdbcClient jdbcClient, long refreshPeriod, String initQuery,
+                                      String updateQuery, TimeoutFactory timeoutFactory, long timeout) {
+        this.cacheNotificationListener = Objects.requireNonNull(cacheNotificationListener);
+        this.vertx = Objects.requireNonNull(vertx);
+        this.jdbcClient = Objects.requireNonNull(jdbcClient);
+        this.refreshPeriod = refreshPeriod;
+        this.initQuery = Objects.requireNonNull(StringUtils.stripToNull(initQuery));
+        this.updateQuery = Objects.requireNonNull(StringUtils.stripToNull(updateQuery));
+        this.timeoutFactory = Objects.requireNonNull(timeoutFactory);
+        this.timeout = timeout;
+    }
+
+    public void initialize() {
+        getAll();
+        if (refreshPeriod > 0) {
+            vertx.setPeriodic(refreshPeriod, aLong -> refresh());
+        }
+    }
+
+    private void getAll() {
+        lastUpdate = Instant.now();
```

Comment:
Pls wrap setting `lastUpdate` in separate method like 
```
Void setLastUpdate(Instant instant) {
  lastUpdate = instant;
  return null;
}
```
So, we'll ensure in consistency of `lastUpdate` in case of any error occurred.

PS. use the same call in `refresh()`.